### PR TITLE
[receiver/mongodb] Respect TLS in all situations

### LINKED ISSUE #12092: [receiver/mongodb] TLS Settings Ignored in certain circumstances
**Describe the bug**
Did some digging and noticed the alpha `receiver/mongodbreceiver` was checking for a non-nil error from `configtls.LoadTLSConfig()` before inputting into the `mongo-go-driver`

**Steps to reproduce**
Set up a mongodb 4.4 instance with these settings:

```yaml
net:
   tls:
      mode: requireTLS
      certificateKeyFile: /opt/mongo/tls/mongo-test.localnet.pem
      allowConnectionsWithoutCertificates: true
```

And the mongod logs indicate a problem from the client:
```
{"t":{"$date":"2022-07-01T22:21:27.345+00:00"},"s":"I",  "c":"NETWORK",  "id":22988,   "ctx":"conn8528","msg":"Error receiving request from client. Ending connection from remote","attr":{"error":{"code":141,"codeName":"SSLHandshakeFailed","errmsg":"The server is configured to only allow SSL connections"},"remote":"127.0.0.1:58444","connectionId":8528}}
```

**What did you expect to see?**
Before: 
```go
        tlsConfig, err := c.LoadTLSConfig()
	if err != nil && tlsConfig != nil {
		clientOptions.SetTLSConfig(tlsConfig)
	}
```

After:
```go
        tlsConfig, err := c.LoadTLSConfig()
	if err == nil && tlsConfig != nil {
		clientOptions.SetTLSConfig(tlsConfig)
	}
```

**What did you see instead?**
Metrics collection did work through TLS

**What version did you use?**
Version 0.54.0 version of the `mongodbreceiver`

**What config did you use?**
```yaml
receivers:
  mongodb:
    hosts:
      - endpoint: keith-mongo-test.bluemedora.localnet:27017
    username: admin
    password: password
    collection_interval: 20s
    tls:
      insecure: false
      cert_file: /opt/mongo/tls/mongo-test.bluemedora.localnet.crt
      key_file: /opt/mongo/tls/mongo-test.bluemedora.localnet.key
      insecure_skip_verify: true
exporters:
  logging:

service:
  pipelines:
    metrics:
      receivers:
      - mongodb
      exporters:
      - logging
[root@keith-mon
```

**Environment**
OS:  "RHEL 7.9"

**Additional context**

Will author a PR shortly to resolve this!

### PR DESCRIPTION
**Pull request by schmikei**
**Description:** <Describe what has changed.>
Resolves #12092  

**Link to tracking Issue:** <Issue number if applicable>
#12092

**Testing:** <Describe what testing was performed and which tests were added.>
Manually tested/working on unit test

**Documentation:** unreleased yaml file added for bug fix

### CODE REVIEW

**Comment by schmikei** [comment]
@djaglowski do you mind taking a look at this?

File: receiver/mongodbreceiver/testdata/certs/ca-1.cert [hunk 1]
```diff
@@ -0,0 +1,20 @@
+-----BEGIN CERTIFICATE-----
```
Comment (djaglowski) [inline receiver/mongodbreceiver/testdata/certs/ca-1.cert:1]: We have tooling to autogenerate certs. Is it possible to hook this into the [Makefile](https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/0a6d3531e4d4e992b4936f7a9a4b55092a41efc4/Makefile#L356)?

**Review by djaglowski — COMMENTED** [review]
Looks good - just one question.

File: receiver/mongodbreceiver/testdata/certs/ca-1.cert (hunk 1 above)
Comment (schmikei) [reply receiver/mongodbreceiver/testdata/certs/ca-1.cert:1]: Ahh good catch, also noticed a typo in the Makefile or directory structure for mongodbatlas: 

`receiver/mongodbatlasreceiver/testdata/alerts/certs` should be => `receiver/mongodbatlasreceiver/testdata/alerts/cert`

I can file a separate issue or I can fix it in this PR but just noticed, do you have a preference on which approach?

Comment (djaglowski) [reply receiver/mongodbreceiver/testdata/certs/ca-1.cert]: Nice catch. Do you mind creating a separate PR?

Comment (schmikei) [reply receiver/mongodbreceiver/testdata/certs/ca-1.cert]: Yep accidentally committed to this branch, I'll fix that change

File: Makefile [hunk 2]
```diff
@@ -356,7 +356,8 @@ endef
 CERT_DIRS := receiver/sapmreceiver/testdata \
              receiver/signalfxreceiver/testdata \
              receiver/splunkhecreceiver/testdata \
-             receiver/mongodbatlasreceiver/testdata/alerts/certs
+             receiver/mongodbatlasreceiver/testdata/alerts/certs \
```
Comment (djaglowski) [inline Makefile:359]:
```suggestion
             receiver/mongodbatlasreceiver/testdata/alerts/cert \
```
Since this is being handled in another PR

Comment (schmikei) [reply Makefile:359]: That would be the fix in #12118? As currently written, the issue for that would still be open right?

**Review by djaglowski — APPROVED** [review]

**Pull request merged**